### ISSUE TITLE
Issue: Still showing [native code] even with source maps

### ISSUE THREAD

**Issue by hellsan631** [expo/sentry-expo#74]
Really scratching my head here. I am unable to get source maps to correctly work in a provisioned version of sentry, using the latest version of expo & react native.

Releases is correctly set, sentry server version is 9.1.2,
expo: "34.0.4", sentry-expo: "2.0.0",
```
import * as Sentry from 'sentry-expo';
import Constants from 'expo-constants';

const dsn = 'Something';
const { revisionId: release } = Constants.manifest;

Sentry.init({
  dsn,
  environment: __DEV__ ? 'development' : 'production',
  release,
  enableInExpoDevelopment: true,
  debug: true
});

Sentry.setDist(Constants.nativeBuildVersion); // This doesn't seem to do anything helpful?
Sentry.setRelease(release);

export default Sentry;
```

Using the post took to upload our source maps, we are still seeing "native code"

![Screen Shot 2019-09-12 at 8 14 28 PM](https://user-images.githubusercontent.com/1474758/64829352-f9d96f80-d599-11e9-84e9-ae1a3a0b6725.png)

Even though this is thrown inside javascript, with build artifacts existing for that release.

![Screen Shot 2019-09-12 at 8 16 25 PM](https://user-images.githubusercontent.com/1474758/64829417-3d33de00-d59a-11e9-8604-8eb2811ed92d.png)

**Comment by bill-pairaktaridis** [comment]
I'm having the same issue. However, it's not happening everywhere. I think it may be connected to the connect() function of Redux. Are you also using it?

**Comment by hellsan631** [comment]
@bill-pairaktaridis  yes.

This is also happening in more isolated areas as well. functional components that do not connect to a store (within their scope), but may well be imported by components that use connect.

We are currently trying to remove redux, but that is a few months away from being complete.

**Comment by bill-pairaktaridis** [comment]
Yes, maybe one of the sentry-expo devs can help out here. In its current form it offers little in the way of actually debugging an issue.

**Comment by b4stien** [comment]
I'm under the impression that it's a prefix issue. My code is prefixed with `app:///` vs the uploaded sourcemaps that are in `~/`. I have an explicit warning telling me Sentry has not found my sourcemaps. Cf screenshots.

<img width="789" alt="Screenshot 2019-09-18 at 13 31 13" src="https://user-images.githubusercontent.com/1778107/65145137-1c93da00-da19-11e9-9434-cd287d3f27a1.png">

<img width="1061" alt="Screenshot 2019-09-18 at 13 35 37" src="https://user-images.githubusercontent.com/1778107/65145193-3d5c2f80-da19-11e9-9cc6-3624fc9f991c.png">

**Comment by b4stien** [comment]
Pinging @HazAT as he wrote the `RewriteFrame` for `app:///` (sorry to disturb if you're not the right person to ask).

**Comment by hellsan631** [comment]
When i tried to use the base `sentry-native` repo, it added the `app://` prefix, but didnt solve that problem.

I do not have any warnings about sourcemaps.

It also looks like your code is getting some bundle information (onPress, touchableHandlePress), where as mine is just ending at `[native code]`

**Comment by bill-pairaktaridis** [comment]
@hellsan631 I'll try adding `sentry-native` to my project and see how it goes. As it stands right now, I'm not getting any meaningful stack trace within my components/screens

**Comment by jacinto123** [comment]
I am having this issue as well, anyone got any ideas?

**Comment by bill-pairaktaridis** [comment]
Perhaps passing `--finalize` and/or `--validate` to the executed command might help

See: https://docs.sentry.io/platforms/javascript/sourcemaps/

**Comment by bill-pairaktaridis** [comment]
It might be an issue with `sentry-react-native` itself

See: https://github.com/getsentry/sentry-react-native/issues/601

**Comment by bill-pairaktaridis** [comment]
@cruzach could you look into this? It looks like updating to `sentry-react-native 1.0.6` would solve this. I tried to update on my end by the uploads break.

**Comment by bill-pairaktaridis** [comment]
I've forked `sentry-expo` and updated to `"@sentry/react-native": "^1.0.6"` but it's still happening. You can see my fork here: https://github.com/bill-pairaktaridis/sentry-expo

**Comment by cruzach** [comment]
Could you post the output of running `EXPO_DEBUG=true expo publish`? that might shed a little more light on what's going wrong with uploading the source maps

Also, a way for me to reproduce this would help. I'm getting source maps using `"sentry-expo": "^2.0.0"`

**Comment by cruzach** [comment]
@b4stien I just got the same error shown on the Sentry dashboard, it's due to the `release` missing. Can you reinstall `sentry-expo` (made a fix where the release will then be auto-configured on `Sentry.init`) , and try again?

**Comment by b4stien** [comment]
@cruzach it works indeed, thank you very much!

**Comment by bill-pairaktaridis** [comment]
I don't have any errors in my issues. But the code is still showing up minified and with no mapping to the source. I distinctly remember this working for a while (for me) before the upgrade to the unified Sentry JS SDK.

**Comment by bill-pairaktaridis** [comment]
@cruzach I'm still seeing it. I'll try to set up a clean Expo project and test it on that. Could you also do a rudimentary Redux setup and see if that's the cause? Because right now, this is basically what all my issues look like.
<img width="451" alt="Screenshot 2019-10-01 at 10 58 04 AM" src="https://user-images.githubusercontent.com/1397468/65944457-65e12200-e43a-11e9-89ed-1935d3c905ec.png">

**Comment by bill-pairaktaridis** [comment]
@hellsan631 did you get it to work?

**Comment by bill-pairaktaridis** [comment]
@cruzach did you try with a built app (I'm using App Bundles, as that's mandatory now for the Google Play Store)?

**Comment by bill-pairaktaridis** [comment]
@cruzach I'm seeing that in files that don't contain `connect()` or `react-navigation` my errors get reported correctly. So it must be one or the other.

**Comment by cruzach** [comment]
I don't believe it's `react-navigation` related, got a project with the most recent `sentry-expo` working implementing that, already. 

I'll try and repro this, if you have time to put together a minimal example using redux that demonstrates this problem, feel free to link a repro

**Comment by bill-pairaktaridis** [comment]
I'm almost positive it has something to do with `connect()` or in general something that is in my screens.  I've had errors pop up in my newly implemented "models" and they are properly presented. I'll comment back when I have newer issues and we can maybe update the docs to reflect this case, as Redux is promoted by Expo themselves.

**Comment by bill-pairaktaridis** [comment]
I was able to resolve this! The fault was on my end because I was catching all errors on my AppScreen.js with `componentDidCatch`. Apparently the way I did it wasn't including breadcrumbs. I removed that and now errors are presented correctly, even though I'd like to be able to centrally handle all errors.

**Comment by hellsan631** [comment]
@bill-pairaktaridis how did you catch breadcrumbs as well?

**Comment by bill-pairaktaridis** [comment]
@hellsan631 I didn't. I just removed my `componentDidCatch` and let Sentry handle it. There's definitely a way to do it but in our current development cycle I was just happy to get it working and being able to debug again.

**Comment by alexanderritola** [comment]
Sentry docs appear to show how to resolve this.

https://docs.sentry.io/platforms/javascript/react/#error-boundaries

**Comment by bill-pairaktaridis** [comment]
Unfortunately, this issue seems to have resurfaced. This is the screen I'm looking at again:

<img width="495" alt="Screenshot 2020-03-30 at 10 20 55 AM" src="https://user-images.githubusercontent.com/1397468/77885932-a2b69600-7270-11ea-8c16-170f064600f5.png">

I'm pretty sure this error occurs within a fetch statement, if that is any indication

**Comment by cruzach** [comment]
i think this occurs only for unhandled promise rejections on android

see: 
[sync func call stacktrace](https://sentry.io/share/issue/6c8fe2b5712740c482041ed43d764d97/)

[async func call stacktrace](https://sentry.io/share/issue/35ad0b19939346f5a6883c9ec1f16a3e/)

@HazAT curious if this is an issue specific to `sentry-expo` or if it could be a sentry or general react native issue?

**Comment by bill-pairaktaridis** [comment]
@cruzach yes that's what I've surmised as well. I keep getting a couple every day that are all promise rejections that I'm apparently not handling somewhere in the code. Hard to troubleshoot though without any stack trace.

**Comment by cruzach** [comment]
i think this is a duplicate of https://github.com/expo/sentry-expo/issues/4

**Comment by skoch-tf** [comment]
check if your source map was published with success:
> expo publish
```
Running postPublish hook: sentry-expo/upload-sourcemaps
Created release 1.3.112-r0-CX15Bte!.
```

**Comment by llamaluvr** [comment]
Heads up that I see this same behavior when I call `Sentry.captureException()` within a MobX State Tree middleware and in a `componentDidCatch` on my root component. I see a few `[native code]` blocks on iOS stack traces, depending on the error, but usually get at least some partial sense of where the error is. However, I get almost no useful info out of Android errors. I just put in a support request to Sentry to see if they had any ideas.

**Comment by juspeeh** [comment]
Also facing similar issues.
expo: 39.0.4
sentry-expo: 3.0.3

Source maps have been uploaded successfully.

<img width="854" alt="Screenshot 2021-02-15 at 17 24 05" src="https://user-images.githubusercontent.com/25384349/107965311-0cd44880-6fb3-11eb-81a5-cae531031197.png">

**Comment by juspeeh** [comment]
@cruzach

Any information I can provide to solve this issue? Or any advice I should check on my environment that could cause this issue?

All the errors I'm getting are filled with [native code] and I'm not able to pinpoint any errors from the production environment.

**Comment by cruzach** [comment]
I believe this was an upstream sentry issue (https://github.com/getsentry/sentry-react-native/issues/1077) and was just recently fixed. Once a new release is published, you'll just need to remove your node modules and lockfile and reinstall sentry-expo, since it should automatically pull in the newest version of `@sentry/react-native` (unless it's a breaking change)

**Comment by cruzach** [comment]
Newest version of `@sentry/react-native` was released- 2.2.2
